platform/ios/app-swift/Sources/ManageOfflineRegionsExample.swift

```diff
@@ -0,0 +1,238 @@
+         increases, you should be conservative with your `toZoomLevel` setting.
+         */
+        let region = MLNTilePyramidOfflineRegion(styleURL: mapView.styleURL,
+                                                 bounds: mapView.visibleCoordinateBounds,
+                                                 fromZoomLevel: mapView.zoomLevel,
+                                                 toZoomLevel: mapView.zoomLevel + 2)
+        // Store some data for identification purposes alongside the offline pack.
+        let userInfo = UserData(name: "\(region.bounds)")
+        let jsonEncoder = JSONEncoder()
+        let optionalContext = try? jsonEncoder.encode(userInfo)
+        guard let context = optionalContext else {
+            print("Error: failed to get context")
+            return
+        }
+
+        // Create and register an offline pack with the shared offline storage object.
+        MLNOfflineStorage.shared.addPack(for: region, withContext: context) { pack, error in
+            guard error == nil else {
+                // Handle the error if the offline pack couldn’t be created.
+                print("Error: \(error?.localizedDescription ?? "unknown error")")
+                return
+            }
+
+            // Begin downloading the map for offline use.
+            pack!.resume()
+        }
+    }
+
+    // MARK: - MLNOfflinePack notification handlers
+
+    @objc func offlinePackProgressDidChange(notification: NSNotification) {
+        /**
+         Get the offline pack this notification is referring to,
+         along with its associated metadata.
+         */
+        if let pack = notification.object as? MLNOfflinePack,
+           let userInfo = try? jsonDecoder.decode(UserData.self, from: pack.context)
+        {
+            // At this point, the offline pack has finished downloading.
+
+            if pack.state == .complete {
+                let byteCount = ByteCountFormatter.string(fromByteCount: Int64(pack.progress.countOfBytesCompleted), countStyle: ByteCountFormatter.CountStyle.memory)
+
+                let packName = userInfo.name
+
+                print("""
+                  Offline pack “\(packName)” completed download:
+                    - Bytes: \(byteCount)
+                    - Resource count: \(pack.progress.countOfResourcesCompleted)")
+                """)
+
+                NotificationCenter.default.removeObserver(self, name: NSNotification.Name.MLNOfflinePackProgressChanged,
+                                                          object: nil)
+            }
+        }
+        // Reload the table to update the progress percentage for each offline pack.
+        tableView.reloadData()
+    }
+}
+
+private extension MLNOfflinePackProgress {
+    var percentCompleted: Float {
+        guard countOfResourcesExpected != 0 else {
+            return 0
+        }
+
+        let percentage = Float((countOfResourcesCompleted / countOfResourcesExpected) * 100)
```

Comment:
```suggestion
        let percentage = Float(countOfResourcesCompleted) / Float(countOfResourcesExpected) * 100
```

I'm type checking in my head, but I _think_ both of the values in the division are unsigned integers (source: https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/mlnofflinepackprogress). So I think the result of this division will always be either zero or one (Swift doesn't implicitly convert integers to floating point values).

Reply:
You are completely right and nice catch.